utils/uiTypes/VotePlugin.ts

```diff
@@ -324,94 +518,38 @@ export class VotingClient {
             new AccountData(nftVoteRecord, false, true)
           )
       }
+
       //1 nft is 3 accounts
       const firstFiveNfts = remainingAccounts.slice(0, 15)
-      const remainingNftsToChunk = remainingAccounts.slice(
-        15,
-        remainingAccounts.length
-      )
-      const nftsChunk = chunks(remainingNftsToChunk, 12)
-      for (const i of nftsChunk) {
-        const castNftVoteIx = await this.client.program.methods
-          .castNftVote(proposal.pubkey)
-          .accounts({
-            registrar,
-            voterWeightRecord: voterWeightPk,
-            governingTokenOwner: walletPk,
-            payer: walletPk,
-            systemProgram: SYSTEM_PROGRAM_ID,
-          })
-          .remainingAccounts(i)
-          .instruction()
-
-        instructions.push(castNftVoteIx)
-      }
-      const castNftVoteIx2 = await this.client.program.methods
-        .castNftVote(proposal.pubkey)
-        .accounts({
-          registrar,
-          voterWeightRecord: voterWeightPk,
-          governingTokenOwner: walletPk,
-          payer: walletPk,
-          systemProgram: SYSTEM_PROGRAM_ID,
-        })
-        .remainingAccounts(firstFiveNfts)
-        .instruction()
-      instructions.push(castNftVoteIx2)
-      return { voterWeightPk, maxVoterWeightRecord }
-    }
-
-    if (this.client instanceof VsrClient) {
-      const props = await this.withUpdateVoterWeightRecord(
-        instructions,
-        tokeOwnerRecord,
-        'castVote'
-      )
-      return props
-    }
-    if (this.client instanceof SwitchboardQueueVoterClient) {
-      const props = await this.withUpdateVoterWeightRecord(
-        instructions,
-        tokeOwnerRecord,
-        'castVote'
+      const remainingNftsChunk = chunks(
```

Comment:
Yeah my apologies, Since our implementation of vsr uses NFTs I used the current implementation of NFTVoterClient and mirrored a-lot of the logic. When it came to this function it seemed redundant to just call the same method on the program `castNftVoteIx2 ` when there was already a loop above it going over the chunks and calling that instruction. So I just combined them with for `(const chunk of [firstFiveNfts, ...remainingNftsChunk])`. The testing I would say would be done through our vsr instance as its using NFTs. I can go ahead and create a mock realms instance with some fake nfts to test if you want.

Reply:
I agree seems redundant, the original code is just hard to follow and I dont feel confident in my ability to audit. Testing would be good